src/main.rs

```diff
@@ -55,25 +59,25 @@ async fn main() -> anyhow::Result<()> {
         command,
     } = Args::parse();
 
+    // Expand vector as more CRDs are created.
+    let crds = vec![kubit::resources::AppInstance::crd()];
     match &command {
-        Some(Commands::Manifests { crd_dir }) => match crd_dir {
-            Some(crd_dir) => {
-                // Expand vector as more CRDs are created.
-                for crd in vec![&kubit::resources::AppInstance::crd()] {
-                    let crd_file = format!("{}_{}.yaml", crd.spec.group, crd.spec.names.plural);
-                    let crd_path = PathBuf::from(crd_dir).join(crd_file);
-
-                    let file =
-                        File::create(crd_path).expect("Could not open AppInstances CRD file");
-
-                    serde_yaml::to_writer(&file, &kubit::resources::AppInstance::crd())?;
-                }
+        Some(Commands::Manifests { crd_dir }) => {
+            for crd in crds {
+                let mut out_writer: Box<dyn Write> = match crd_dir {
+                    Some(dir) => {
+                        let crd_file = format!("{}_{}.yaml", crd.spec.group, crd.spec.names.plural);
+                        let file = File::create(dir.join(crd_file))
+                            .expect("Could not open AppInstances CRD file");
+                        Box::new(file)
+                    }
+                    None => Box::new(stdout()),
+                };
+                // The YAML delimiter is added in the event we have multiple documents.
+                out_writer.write(b"---\n").unwrap();
```

Comment:
We should add a linting step, since it reveals interesting things:

```
error: written amount is not handled
  --> src/main.rs:77:17
   |
77 |                 out_writer.write(b"---\n").unwrap();
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `Write::write_all` instead, or handle partial writes
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_io_amount
   = note: `#[deny(clippy::unused_io_amount)]` on by default
```

anyway, it's more idiomatic to use `writeln!(...)`